rpc/handlers.go

```diff
@@ -1505,14 +1505,24 @@ func (h *Handler) TraceTransactionV0_6(ctx context.Context, hash felt.Felt) (*vm
 }
 
 func (h *Handler) traceTransaction(ctx context.Context, hash *felt.Felt, v0_6Response bool) (*vm.TransactionTrace, *jsonrpc.Error) {
-	_, _, blockNumber, err := h.bcReader.Receipt(hash)
+	_, blockHash, _, err := h.bcReader.Receipt(hash)
 	if err != nil {
 		return nil, ErrTxnHashNotFound
 	}
 
-	block, err := h.bcReader.BlockByNumber(blockNumber)
-	if err != nil {
-		return nil, ErrBlockNotFound
+	var block *core.Block
+	pendingBlock := blockHash == nil
+	if pendingBlock {
+		pending, err := h.bcReader.Pending()
+		if err != nil {
+			return nil, ErrInternal
```

Comment:
The reason it's internal and not `BlockNotFound` is because afaik pending block should be always present and if it's not there are some other problems how we deal with it. @IronGauntlets right?

Reply:
We checked with Aneeque: these handlers should return only TxHashNotFound errors